standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/Warehouse.java

```diff
@@ -355,7 +355,11 @@ public Path getDefaultTablePath(Database db, String tableName, boolean isExterna
     Path dbPath = null;
     if (isExternal) {
       dbPath = new Path(db.getLocationUri());
-      if (FileUtils.isSubdirectory(getWhRoot().toString(), dbPath.toString() + Path.SEPARATOR)) {
+      Path dbLocation = Path.getPathWithoutSchemeAndAuthority(dbPath);
```

Comment:
This is a common API thats used by other areas of the code in Hive. I am concerned that doing a scheme-less path comparison might cause issues.
dbPath is the location set on the Database object which has the scheme. for ex: s3 or hdfs.
if we do a scheme-less comparison, s3://user/warehouse/hive/mydb.db/mytable will be considered a managed location if the warehouse root is set to "hdfs://user/warehouse/hive". 
Do you see a downside with just toString comparison?

Reply:
Good catch ! 
We shouldn't do schema less comparison here. Will fix it.